orchestra/static/orchestra/project-management/data-service.es6.js

```diff
@@ -43,15 +42,13 @@ export default function dataService ($location, $rootScope, $route, orchestraApi
         })
     },
     changeProject: function (projectId) {
-      this.ready.then(function () {
-        if (!projectId || !this.allProjects[projectId]) {
-          $route.updateParams({projectId: undefined})
-          return
-        }
-
-        this.currentProject = this.allProjects[projectId]
-        return this.setSelectedProject()
-      }.bind(this))
+      if (!projectId) {
+        $route.updateParams({projectId: undefined})
+        return
+      }
+      this.currentProject.id = projectId
+      this.resetData()
+      return this.updateData()
```

Comment:
curiously, why remove `this.ready` to make sure we have the data, why remove `!this.allProjects[projectId]`, and why change the selection logic I've highlighted?